### PR TITLE
Fix validate_resource script

### PR DESCRIPTION
**Pull request by myakove**
## Summary by CodeRabbit

- **New Features**
	- Introduced a validation tool for Kubernetes resource definitions, ensuring compliance with expected API standards.
	- Added a comprehensive suite of unit tests for validating resource class structures and correctness.

- **Bug Fixes**
	- Streamlined the process for gathering resource files, enhancing the reliability of resource validation.

- **Refactor**
	- Reorganized validation logic and testing strategy for improved maintainability and clarity in resource validation processes.

- **Chores**
	- Added new test environments to enhance testing capabilities without altering existing functionalities.

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: tests/scripts/validate_resources.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+    base_class_from = [_base for _base in bases if _base.id in (resource_str, namespaced_resource_str)]
+    if not base_class_from:
+        return [f"{_base_class_error}, got {[_base.id for _base in bases]}"]
+
+    if len(base_class_from) > 1:
+        return [f"Resource {cls.name} must have only one of {resource_str} or {namespaced_resource_str} base class"]
+
+    if not api_type:
+        return [f"Resource {cls.name} must have api_group or api_version"]
```
Comment (rnetser) [inline tests/scripts/validate_resources.py:62]: we do not want to check base class as well?

Comment (myakove) [reply tests/scripts/validate_resources.py:62]: What do you mean?

Comment (rnetser) [reply tests/scripts/validate_resources.py:64]: check if base class is the right one is done after this part

**Review by rnetser — APPROVED** [review]

**Pull request merged**